### PR TITLE
Prevent mailto links from being joined with prefix

### LINKED ISSUE #228: mailto links in README.md don't work with "vsce package"
Repro steps:
1. Add mailto link in README.md, e.g. \[Email me!](mailto:me@website.com?Subject=Repro).
2. Run "vsce package"
3. I get the following error:
Error: Couldn't detect the repository where this extension is published. The link 'mailto:me@website.com?Subject=Repro' will be broken in README.md. Please provide the repository URL in package.json or use the --baseContentUrl and --baseImagesUrl options.

If I run "vsce package --baseContentUrl" it ends up adding "true/" to the link like so:
\[Email me!](true/mailto:me@website.com?Subject=Repro)

I would expect vsce to ignore mailto links such that they actually work when the package is installed and the user views the readme.

### PR DESCRIPTION
**Pull request by JeffreyCA**
Resolves #228

Updated some tests and they all pass.

### CODE REVIEW

File: src/package.ts [hunk 1]
```diff
@@ -390,13 +390,14 @@ export class MarkdownProcessor extends BaseProcessor {
 		}
 
 		const markdownPathRegex = /(!?)\[([^\]\[]*|!\[[^\]\[]*]\([^\)]+\))\]\(([^\)]+)\)/g;
-		const urlReplace = (_, isImage, title, link) => {
+		const urlReplace = (_, isImage, title, link: string) => {
 			const isLinkRelative = !/^\w+:\/\//.test(link) && link[0] !== '#';
+			const isMailTo = link.toLowerCase().startsWith('mailto:');
```
Comment (joaomoreno) [inline src/package.ts:395]:
```suggestion
			if (/^mailto:/i.test(link)) {
				return link;
			}
			
			const isLinkRelative = !/^\w+:\/\//.test(link) && link[0] !== '#';
```

File: src/package.ts [hunk 2]
```diff
@@ -390,13 +390,14 @@ export class MarkdownProcessor extends BaseProcessor {
...
 
 			if (!this.baseContentUrl && !this.baseImagesUrl) {
 				const asset = isImage ? 'image' : 'link';
 
-				if (isLinkRelative) {
+				if (isLinkRelative && !isMailTo) {
```
Comment (joaomoreno) [inline src/package.ts:400]:
```suggestion
				if (isLinkRelative) {
```

File: src/package.ts [hunk 3]
```diff
@@ -408,7 +409,8 @@ export class MarkdownProcessor extends BaseProcessor {
 				return `${isImage}[${title}](${link})`;
 			}
 
-			return `${isImage}[${title}](${urljoin(prefix, link)})`;
+			const finalLink = isMailTo ? link : urljoin(prefix, link);
+			return `${isImage}[${title}](${finalLink})`;
```
Comment (joaomoreno) [inline src/package.ts:413]:
```suggestion
			return `${isImage}[${title}](${urljoin(prefix, link)})`;
```

File: src/package.ts (hunk 1 above)
Comment (JeffreyCA) [reply src/package.ts]: Thanks for the suggestions. I believe this should return `${isImage}[${title}](${link})` instead.

**Comment by joaomoreno** [comment]
Thanks!

**Pull request merged**